src/window.rs

```diff
@@ -132,6 +146,48 @@ impl Window {
         window
     }
 
+    /// Create the control panel
+    fn create_control_panel(
+        window: &winit::window::Window,
+        webview_id: Option<WebViewId>,
+    ) -> WebView {
+        let size = window.inner_size();
+        let size = Size2D::new(size.width as i32, size.height as i32);
+        if let Some(webview_id) = webview_id {
+            WebView::new(webview_id, DeviceIntRect::from_size(size))
+        } else {
+            WebView::new_panel(DeviceIntRect::from_size(size))
+        }
+    }
+
+    /// Get the content area size for the webview to draw on
+    pub fn get_content_rect(&self, mut size: DeviceIntRect) -> DeviceIntRect {
+        if self.panel.is_some() {
+            size.min.y = size.max.y.min(100);
+            size.min.x += 10;
+            size.max.y -= 10;
+            size.max.x -= 10;
+        }
+        size
+    }
+
+    /// Send the constellation message to start Panel UI
+    pub fn init_panel_webview(
```

Comment:
Is it possible to move this into window's new methods or create_control_panel?

Reply:
That means we'll need to pass in `resource_dir` and `constellation_sender` in `new`

Reply:
hmmm let's discuss this in zulip.